docs/api.md

```diff
@@ -0,0 +1,191 @@
+# Go API to Run Flogo Application
+
+## Run Flow
+This short tutorial walks through how to run Flogo Actions using the Go-API provided by the flogo core.
+
+First Import the action you need to run .
+
+```go 
+import (
+    ...
+    "github.com/project-flogo/core/api"
+    "github.com/project-flogo/flow"
+    ...
+)
+```
+
+Then initalize the Flogo app to use Go-API
+
+```go
+app := api.NewApp()
+```
+
+Setup the trigger and handler that invokes the action.   In this case REST trigger is selected to run on port 8080. GET "/blah/:value" (on REST trigger) is registered as a handler.
+```go
+
+trg := app.NewTrigger(&rest.Trigger{}, &rest.Settings{Port: 8080})
+h, _ := trg.NewHandler(&rest.HandlerSettings{Method: "GET", Path: "/blah/:num"})
+
+```
+
+Now setup Flow Action that runs the flow from local file myflow.json. Then associate that action with our handler.
+```go
+
+settings :=  &flow.Settings{FlowURI:"file://myflow.json"}
+a, _ h.NewAction(&flow.FlowAction{}, settings)
+
+```
+
+Now lets map the path parameter from the handler to input of our flow.
+```go
+
+a.SetInputMappings("in=$.pathParams.val")
+
+```
+
```

Comment:
The following sections are a little confusing.  I would make each example a full go file littered with comments and a full paragraph explaining it before the example